swaylock/main.c

```diff
@@ -249,7 +283,38 @@ int main(int argc, char **argv) {
 			break;
 		}
 		case 'i':
-			image_path = optarg;
+			if(images.type == NOL_NONE) {
+				images = (struct none_one_list) { NOL_ONE, optarg, NULL };
+			} else {
+				fprintf(stderr, "only one of --image/--window-image is allowed\n");
+				exit(EXIT_FAILURE);
+			}
+			break;
+		case 'w':
+			if(images.type == NOL_NONE) {
+				images = (struct none_one_list) { NOL_LIST, create_list(), create_list() };
+				for(i = 0; i <= ((list_t*) images.a)->capacity; ++i) {
+					// this isn't perfect - if you hav>e more than 10 monitors there's a small chance of
+					// a segfault from uninitialized data. I honestly don't think enough people have more than
+					// 10 monitors to care about that though.
+					((list_t*) images.a)->items[i] = NULL;
+					((list_t*) images.b)->items[i] = NULL;
+				}
```

Comment:
I think you would be better of using an array if you wan't to encode the output indices in the structure. A list is not really meant for that kind of thing, hence you end up doing all this :)

You could either create a fix sized array, limiting the number of outputs, or you could allocate it from `registry->outputs->length`.